service/history/shard/events_cache.go

```diff
@@ -24,85 +24,87 @@
 
 //go:generate mockgen -copyright_file ../../../LICENSE -package $GOPACKAGE -source $GOFILE -destination events_cache_mock.go
 
-package events
+package shard
```

Comment:
I had to move this file from events package to shard package. Otherwise, there will be a circular dependency between them. ShardContext has a reference to EventsCache. EventsCache methods take shardContext as an argument.

Open to any comments about better organization.

Reply:
I am not following, why shardContext needs to be an argument here? For logging and metrics tag, if so we can just pass in the shardID. The execution manager is also a host level component and doesn't need to be get from shard context.

Reply:
Yeah, that is my mistake. I somehow assumed that the execution manager is shard level. I have reverted this. Thanks!